warehouse/macros/gtfs_time_string_to_interval.sql

```diff
@@ -0,0 +1,10 @@
+{% macro gtfs_time_string_to_interval(gtfs_time_field) %}
+-- convert a GTFS "Time" string to a BQ interval type
+-- INTERVAL type allows us to handle times past midnight (ex. 26:30:30)
+-- see: https://gtfs.org/schedule/reference/#field-types for how GTFS defines a "Time"
+
+CASE
+    WHEN REGEXP_CONTAINS({{ gtfs_time_field }}, "^[0-9]+:[0-5][0-9]:[0-5][0-9]$") THEN CAST({{ gtfs_time_field }} AS INTERVAL)
```

Comment:
Can you think of any cases where a GTFS time field is potentially negative? And if negatives are possible source values here, does this properly cast to the INTERVAL in a way that distributes the negative across hour/minute/second values?

Reply:
Times are not supposed to be negative. GTFS doesn't officially call these intervals; they portray it as like an extension of `HH:MM:SS` rather than an interval, we are just using the BigQuery interval type because it natively handles `HH > 24`. 
<img width="1426" alt="image" src="https://user-images.githubusercontent.com/55149902/236905932-fff057e0-15f9-45f3-b7f3-5f8add1857be.png">